components/api/src/test/java/com/hotels/styx/api/extension/service/spi/MockContext.java

```diff
@@ -33,4 +37,9 @@ public void add(String key, Object value) {
     public boolean isSecure() {
         return false;
     }
+
+    @Override
+    public Optional<InetSocketAddress> clientAddress() {
+        return Optional.empty();
+    }
```

Comment:
Hi @kvosper I notice `InetSocketAddress` is an Optional. Under what circumstances it is absent? Can you remind me? Apart from unit tests I cannot think of any situation where client address wouldn't be available in the Context.

Reply:
When we use the Styx client, there is a line `host.hostClient().handle(request, new HttpInterceptorContext())`. This requires a context because `hostClient` returns an `HttpHandler`, however `clientAddress` is not available in `StyxHttpClient` (nor it is relevant).

Reply:
I see.